collated/main/A0111277M.md

```diff
@@ -0,0 +1,594 @@
+        	
+            return new Event (id, name, description, startTime, endTime, tag, doneStatus) ;
+        
+        } 
+        
+        if (endTime != null && startTime == null) {
+            return new Deadline (id, name, description, endTime, tag, doneStatus) ;
+        
+        } 
+        
+        return new Task (id, name, description, tag, doneStatus) ;
+
+    }
+    
+    /*
+     * Populates the dates map with existing dates there is no such pair.
+     */
+    private void determineDateTimeOfNewTask (ReadOnlyTask taskToEdit) {
+    	
+        if (dateMap.size() > 0) {
+        	hasChanged = true;
+        }
+    	
+        if (taskToEdit instanceof Event) {
+
+            if (!dateMap.containsKey(START_DATE)) {
+                dateMap.put(START_DATE, ( (Event) taskToEdit).getStartDate()) ;
+            } else {
+                hasChanged = true ;
+            }
+            
+            if (!dateMap.containsKey(END_DATE)) {
+                dateMap.put(END_DATE, ( (Event) taskToEdit).getEndDate()) ;
+            } else {
+                hasChanged = true ;
+            }
+
+        }
+
+        if (taskToEdit instanceof Deadline) {
+
+            if (!dateMap.containsKey(END_DATE)) {
+                dateMap.put(END_DATE, ( (Deadline) taskToEdit).getEndDate()) ;
+            } else {
+                hasChanged = true ;
+            }
+        }
+        
+        if(dateMap.containsKey(END_DATE)) {
+            if (dateMap.get(END_DATE).equals(DateUtil.MARKER_FOR_DELETE)) {
+            	dateMap.put(END_DATE, null);
+            } 
+        } else {
+        	dateMap.put(END_DATE, null);
+        }
+
+        
+        if (dateMap.containsKey(START_DATE)) {
+        	if (dateMap.get(START_DATE).equals(DateUtil.MARKER_FOR_DELETE)) {
+        		dateMap.put(START_DATE, null);
+        	} 
+        } else {
+    		dateMap.put(START_DATE, null);
+    	}
+        
+    
+    }
+    
+
+    /* 
+     * Updates the taglist
+     */
+    private Set<Tag> editOrDeleteTags(UniqueTagList currentTags, Set<String> tagNamesSet) throws IllegalValueException {
+        
+        assert model != null ;
+        
+        Set<Tag> newTaskTags = Sets.newHashSet(currentTags) ;
+        
+        for (String names : tagNamesSet) {
+            
+            Tag thisTag = model.getTagRegistry().getTagFromString(names, true) ;
+            
+            if (!currentTags.contains(thisTag)) {
+                newTaskTags.add(thisTag) ;
+                hasChanged = true ;
+                
+            } else {
+                newTaskTags.remove(thisTag) ;
+                hasChanged = true ;
+            }
+            
+        }
+        
+        return newTaskTags ;
+    }
+
+    @Override
+    public Pair<List<ReadOnlyTask>, List<ReadOnlyTask>> getCommandChanges() {
+        return new Pair<List<ReadOnlyTask>, List<ReadOnlyTask>>(ImmutableList.copyOf(tasksAdded), ImmutableList.copyOf(tasksDeleted)) ; 
```

Comment:
Indentation... you can simply use the format code option in Eclipse, Why don't you use it and clean your code?